### PR TITLE
Update advanced blog example

### PR DESCRIPTION
**Pull request by omid**
`structopt` is kinda deprecated

### CODE REVIEW

File: examples/postgres/advanced-blog-cli/Cargo.toml [hunk 1]
```diff
@@ -5,11 +5,11 @@ edition = "2018"
 publish = false
 
 [dependencies]
-bcrypt = "0.10.1"
+bcrypt = "0.14.0"
```
Comment (weiznich) [inline examples/postgres/advanced-blog-cli/Cargo.toml:8]: Could we switch that to [argon2](https://lib.rs/crates/argon2) as that seems to be the state of art password hashing function?

**Review by weiznich — COMMENTED** [review]
Thanks for the PR :+1:
I was aware of that, I just had no time to update that. Would it be possible to switch the password hashing as well?

File: examples/postgres/advanced-blog-cli/Cargo.toml (hunk 1 above)
Comment (omid) [reply examples/postgres/advanced-blog-cli/Cargo.toml:8]: sure

File: examples/postgres/advanced-blog-cli/src/auth.rs [hunk 2]
```diff
@@ -74,7 +82,11 @@ fn register_user(
     username: &str,
     password: &str,
 ) -> Result<User, AuthenticationError> {
-    let hashed_password = hash(password, DEFAULT_COST)?;
+    let salt = SaltString::from_b64(SALT_STRING)?;
```
Comment (weiznich) [inline examples/postgres/advanced-blog-cli/src/auth.rs:85]:
```suggestion
    // In real applications you should never use a constant salt!!
    // Checkout https://cheatsheetseries.owasp.org/cheatsheets/Password_Storage_Cheat_Sheet.html#salting for details
    let salt = SaltString::from_b64(SALT_STRING)?;
```

**Review by weiznich — APPROVED** [review]

**Pull request merged**